app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.tasky.feature_agenda.presentation.agenda_screen
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.feature_agenda.domain.repository.AgendaRepositories
+import com.example.tasky.feature_agenda.domain.use_case.AgendaUseCases
+import com.example.tasky.feature_agenda.presentation.util.generateNextDays
+import com.example.tasky.feature_authentication.domain.util.UserPreferences
+import com.example.tasky.util.Result
+import com.vanpra.composematerialdialogs.MaterialDialogState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val useCases: AgendaUseCases,
+    private val repositories: AgendaRepositories,
+    private val userPreferences: UserPreferences
+): ViewModel() {
+
+    private val _state = MutableStateFlow(AgendaState())
+    val state = _state.asStateFlow()
+
+    var dateDialogState by mutableStateOf(MaterialDialogState())
+
+    var username by mutableStateOf("")
+        private set
+
+    private val resultChannel = Channel<Result<Unit>>()
+    val logoutResult = resultChannel.receiveAsFlow()
+
+    init {
+        val user = userPreferences.getAuthenticatedUser()
+        user?.let {
+            val names = it.fullName.split(" ")
+            username = when(names.size) {
+                1 -> {
+                    "${names[0][0]}${names[0][1]}"
+                }
+
+                2 -> {
+                    "${names[0][0]}${names[1][0]}"
+                }
+
+                else -> ""
+            }
```

Comment:
-What if there's a middle name?
-What if there are 2 spaces between the names? Then splitting based on spaces will give you an empty list in between